Revise libcufile recipe / fix tests.

bdice opened the pull request:
This PR addresses several issues in the libcufile recipe. Changes include:
- `etc/cufile.json` and `man` were not being packaged correctly - these shouldn't go in the `targets` directory as I understand
- Need to skip ppc64le when migrator is enabled (fixed skip logic)
- Added more specific tests for package contents and moved tests to the top level (due to package name conflicts that otherwise prevent tests from running)
- Updated summary/description a bit (description was missing)

Tests are passing locally, so I'll open this PR. However, we have one TODO item that needs addressed: if we intend to package `libcufile_rdma.so` (which is only available on linux64 and not aarch64), we need to figure out how to supply the missing DSOs. See PR for details.

bdice committed:
Fix libcufile tests and packaging.

bdice committed:
Attempt to fix RDMA issue.

bdice committed:
Make tests pass.

jakirkham commented on recipes/libcufile/meta.yaml:
```diff
@@ -20,18 +20,41 @@ source:
+      - targets/{{ target_name }}/lib/libcufile*.so.*
     requirements:
       build:
         - {{ compiler("c") }}
```
```suggestion
        - {{ compiler("c") }}
        - {{ cdt("rdma-core-devel") }}  # [linux64]
```

Edit: Might move this below compilers in the list. Wasn't able to put this change on that part of the diff though

jakirkham commented on recipes/libcufile/meta.yaml:
```diff
@@ -20,18 +20,41 @@ source:
+        # rdma-core-devel-cos7-x86_64 is close, but doesn't seem to work.
+        # Perhaps because the files are in a sysroot location?
+        - "*/libmlx5.so.*"
+        - "*/librdmacm.so.*"
+        - "*/libibverbs.so.*"
```
```suggestion
```

jakirkham commented on recipes/libcufile/meta.yaml:
```diff
@@ -20,18 +20,41 @@ source:
-      run_exports:
-        - {{ pin_subpackage("libcufile", max_pin="x") }}
+        - "*/libcuda.so.*"
+        # TODO: The following libraries are required for libcufile_rdma.so,
+        # which is only for linux64 (not aarch64). What provides these?
+        # rdma-core-devel-cos7-x86_64 is close, but doesn't seem to work.
+        # Perhaps because the files are in a sysroot location?
```
Hopefully the comments below (mostly) address this

If so, we can simplify this comment to explain the CDT (a package that the system will need to provide and Conda won't)

jakirkham commented on recipes/libcufile/meta.yaml:
```diff
@@ -20,18 +20,41 @@ source:
+        # which is only for linux64 (not aarch64). What provides these?
+        # rdma-core-devel-cos7-x86_64 is close, but doesn't seem to work.
+        # Perhaps because the files are in a sysroot location?
+        - "*/libmlx5.so.*"
```
Think this is still outstanding. Can try to look more later today

bdice replied:
This list of three RDMA related files will all be handed by dependencies of rdma-core-devel. No need for any other dependencies, once we figure out the single requirement.

jakirkham replied:
Ah ok. That's right. Forgot this lib was in there

jakirkham reviewed (commented):
Thanks Bradley! 🙏

Added some comment about RDMA/verbs support below

bdice committed:
Fix RDMA cdt dependency.

bdice committed:
Don't ship etc/cufile.json.

bdice committed:
Rerun CI.

bdice commented:
@mroeschke @jakirkham I think all the requests are addressed and this should be ready to merge. I updated the RDMA dependency (thanks for the tips) and also removed `etc/cufile.json` per our offline discussion.

You can see the CI outputs on my fork, which has Azure CI enabled: https://github.com/bdice/staged-recipes/pull/4
